vermouth/processors/do_mapping.py

```diff
@@ -315,20 +322,53 @@ def do_mapping(molecule, mappings, to_ff, attribute_keep=()):
                     graph_out.add_edge(out_idx, out_jdx)
         shared_atoms = set(match1.keys()) & set(match2.keys())
         shared_out_atoms = [mol_to_out[mol_idx] for mol_idx in shared_atoms]
-        for out_atoms in shared_out_atoms:
+        for in_atom, out_atoms in zip(shared_atoms, shared_out_atoms):
             if len(out_atoms) < 2:
+                LOGGER.critical('The atom {} is shared between blocks, but'
+                                ' only mapped once to {}?',
+                                format_atom_string(molecule.nodes[in_atom]),
+                                [format_atom_string(graph_out.nodes[idx])
+                                 for idx in out_atoms],
+                                type='inconsistent-data')
                 raise ValueError("This atom is shared between blocks, but only"
                                  " mapped once?")
             for out_idx, out_jdx in combinations(out_atoms, 2):
                 if out_idx != out_jdx:
                     graph_out.add_edge(out_idx, out_jdx)
         if shared_atoms:
-            print("You have a shared atom between blocks. This may mean you"
-                  " have too  particles in your output and/or erroneous bonds.")
+            LOGGER.warning("You have the following atoms that are shared"
+                           " between blocks. This may mean you have too many"
+                           " particles in your output and/or erroneous bonds."
+                           " {}. They've end up in the following output atoms:"
+                           " {}.",
+                           [format_atom_string(molecule.nodes[idx]) for idx in shared_atoms],
+                           [format_atom_string(graph_out.nodes[idx]) for idx in shared_out_atoms],
+                           type='inconsistent-data')
     # TODO: These should be turned into warnings.
```

Comment:
These are not turned into warning. Aren't they?

Reply:
Two of them are. The TODO can be removed.